app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/screens/home/HomeViewController.kt

```diff
@@ -4,57 +4,152 @@ import android.os.Bundle
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.wrapContentSize
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.material.CircularProgressIndicator
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Scaffold
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.material.TopAppBar
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color.Companion
+import androidx.compose.ui.platform.ComposeView
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
 import au.com.gridstone.trainingkotlin.R
+import au.com.gridstone.trainingkotlin.data.Pokemon
 import au.com.gridstone.trainingkotlin.screens.details.DetailViewController
+import au.com.gridstone.trainingkotlin.screens.home.HomeViewState.Failed
+import au.com.gridstone.trainingkotlin.screens.home.HomeViewState.Loading
+import au.com.gridstone.trainingkotlin.screens.home.HomeViewState.Success
+import au.com.gridstone.trainingkotlin.ui.theme.TrainingKotlinTheme
 import com.bluelinelabs.conductor.Controller
 import com.bluelinelabs.conductor.RouterTransaction
 import com.bluelinelabs.conductor.changehandler.HorizontalChangeHandler
-import kotlinx.coroutines.CoroutineScope
-import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.SupervisorJob
-import kotlinx.coroutines.cancelChildren
+import com.google.accompanist.swiperefresh.SwipeRefresh
+import com.google.accompanist.swiperefresh.rememberSwipeRefreshState
 import kotlinx.coroutines.launch
 import org.koin.core.component.KoinComponent
 import org.koin.core.component.get
-import kotlin.coroutines.CoroutineContext
 
-class HomeViewController : Controller(), KoinComponent, CoroutineScope {
+class HomeViewController : Controller(), KoinComponent {
 
   private val viewModel: HomeViewModel = get()
-  override val coroutineContext: CoroutineContext = Dispatchers.Main + SupervisorJob()
 
   override fun onCreateView(
     inflater: LayoutInflater,
     container: ViewGroup,
     savedViewState: Bundle?,
-  ): View = inflater.inflate(R.layout.home_view_controller, container, false)
+  ): View {
+    return ComposeView(container.context).apply {
+      setContent {
+        TrainingKotlinTheme() {
+          PokemonHomeContent(viewModel = viewModel) {
+            router.pushController(
+              RouterTransaction.with(DetailViewController(pokemon = it))
+                .popChangeHandler(HorizontalChangeHandler())
+                .pushChangeHandler(HorizontalChangeHandler())
+            )
+          }
```

Comment:
For events, avoid lifting the lambda out of the invocation. This is because we want to make it clear that this is an event handler. Reserve the final lambda invocation for [content slots](https://chris.banes.dev/slotting-in-with-compose-ui/)
```kotlin
PokemonHomeContent(
  viewModel = viewModel,
  onSelectPokemon = { 
    router.pushController(
      RouterTransaction.with(DetailViewController(pokemon = it))
        .popChangeHandler(HorizontalChangeHandler())
        .pushChangeHandler(HorizontalChangeHandler()
      )
    )
  }
}
```

Reply:
ok sure